src/planner/binder/statement/bind_drop.cpp

```diff
@@ -43,10 +44,35 @@ BoundStatement Binder::Bind(DropStatement &stmt) {
 		stmt.info->schema = entry->schema->name;
 		break;
 	}
-	case CatalogType::DATABASE_ENTRY:
-		// attaching and detaching is read-only
-		stmt.info->catalog = SYSTEM_CATALOG;
+	case CatalogType::DATABASE_ENTRY: {
+		// allow extensions to handle drop database impl
+		auto &base = (DropInfo &)*stmt.info;
+		string database_name = base.name;
+
+		auto &config = DBConfig::GetConfig(context);
+
+		// for now only handling the case where there is one storage extension registered
+		if (config.storage_extensions.empty()) {
+			// use DuckDB default impl
+			// attaching and detaching is read-only
+			stmt.info->catalog = SYSTEM_CATALOG;
+		} else if (config.storage_extensions.size() > 1) {
+			throw BinderException("DROP DATABASE only supported when there is one storage extension registered.");
```

Comment:
Same here - perhaps loop over the storage extensions instead?

Reply:
Done